FWCore/Services/test/AsyncServiceTester.cc

```diff
@@ -212,6 +257,13 @@ namespace edmtest {
         throw cms::Exception("Assert") << "In analyze: status_ was " << status_ << ", expected 1";
       }
       status_ = 0;
+
+      // Wait in the modules's Event transition in the throwing stream
+      // until the module's Event transition has started in some other
+      // stream
+      if (*streamId_ == throwingStream_) {
```

Comment:
This could just be `if ( not waitOnThisStream)` or just in an `else` block.

Reply:
This code is in the `produce()`, whereas the `waitOnThisStream` is defined in `acquire()`. Probably doing the wait on the following line at the end of `acquire()` would be sufficient, but I liked to put it at the end of `produce()`.